Introduce openshift-ci/Makefile to be used on OpenShift CI

pmacik opened the pull request:
There's an issue with the `golangci-lint v1.21.0`'s dependency - `wsd` package.
`wsd` introduced [an API breaking change](https://github.com/bombsimon/wsl/commit/b9c51ff998af696821c48ca6b959ba2a8d3e02d4#diff-676fb336aa3ff093ebaa37c4293f1cd1L67)  in `v1.2.7` that is not fixed in `golangci-lint`'s latest tagged version `v1.21.0` but is fixed in `master` branch.

This PR:
* Introduces downstream `openshift-ci/Makefile` file overriding the upstream `Makefile`
* Fixes the version of `golangci-lint`
* Adds a `lint` make target
* Adds a `unit` make target
* is a dependency for https://github.com/openshift/release/pull/5913

pmacik committed:
Fix golangci-lint version

sbose78 commented:
Good catch.
This would definitely introduce a conflict later on ;)

sbose78 commented:
Do we want to maintain our own copy of Makefile so that when we make fixes there, we don't have issues with merge conflicts later on?

pmacik commented:
/test all

pmacik committed:
Introduce Makefile.redhat file to contain downstream fixes/updates.

pmacik commented:
> Do we want to maintain our own copy of Makefile so that when we make fixes there, we don't have issues with merge conflicts later on?

Good idea @sbose78, I reverted the original `Makefile` and introduced a new `Makefile.redhat` file for our downstream fixes/updates.

So to run our version, let's use `make -f Makefile.redhat ...` commands.

There's a coupled PR to the OpenShift CI repo https://github.com/openshift/release/pull/5913

pmacik committed:
Add 'lint' make target to Makefile.redhat

pmacik commented:
/assign @bamachrn

sbose78 commented:
/assign @akashshinde

akashshinde commented:
/test unit

sbose78 commented:
Is `ci/prow/unit — Job failed.` supposed to pass @pmacik @ldimaggi  ?

pmacik commented:
> Is `ci/prow/unit — Job failed.` supposed to pass @pmacik @ldimaggi ?

Actually, no at this point.

sbose78 commented:
Cool, @akashshinde , so don't worry about it ;)

pmacik commented:
Actually, It looks like the `master` branch needs a new sync with the upstream master (or with the respective 
v3 branch)

akashshinde commented:
@pmacik raised https://github.com/redhat-developer/helm/pull/14 to sync latest master

pmacik committed:
Move Makefile under openshift-ci dir, introduce unit target.
